### PR TITLE
Don't run VersionService.enforce without localStorage

### PR DESCRIPTION
**Pull request by nullflux**
## Related Issue or Background Info

- We think some of these huge exception spikes are due to a combination of a user having a very old UI _and_ having localStorage disabled. This fix should prevent infinite UI reload loops where localStorage is available but disabled. 

## Changes Proposed

- Manually check for localStorage before enforcing the UI version 

## Additional Info

- Recommend viewing this PR with [whitespace disabled](https://github.com/CDCgov/prime-simplereport/pull/2731/files?diff=split&w=1) because there's a big indent block in the tests 

### UI
- [x] Any changes to the UI/UX are approved by design 
- [x] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [x] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [x] Database changes are submitted as a separate PR
  - [x] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [x] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [x] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
  - [x] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [x] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [x] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [x] Any dependencies introduced have been vetted and discussed

## Cloud
- [x] DevOps team has been notified if PR requires ops support
- [x] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

File: frontend/src/app/VersionService.test.tsx [hunk 1]
```diff
@@ -9,70 +9,99 @@ jest.mock("./utils/reload", () => jest.fn());
...
+        process.env.REACT_APP_CURRENT_COMMIT = localSHA;
+        (env as jest.Mock).mockReturnValue("production");
+        (fetch as FetchMock).mockResponseOnce(remoteSHA);
+
+        // WHEN
+        await VersionService.enforce();
+
+        // THEN
+        expect(fetch).not.toHaveBeenCalled();
+        expect(reload).not.toHaveBeenCalled();
```
Comment (nickclyde) [inline frontend/src/app/VersionService.test.tsx:83]: Awesome, great tests!

File: frontend/src/app/VersionService.tsx [hunk 2]
```diff
@@ -6,6 +6,10 @@ import reload from "./utils/reload";
 export const LOCAL_STORAGE_KEY = "mostRecentVersionReload";
 export class VersionService {
   public static async enforce() {
+    if (!this.localStorageIsAvailable()) {
```
Comment (nickclyde) [inline frontend/src/app/VersionService.tsx:9]: ✅

**Review by nickclyde — APPROVED** [review]
Thank you for this!

File: frontend/src/app/VersionService.tsx [hunk 3]
```diff
@@ -43,4 +47,18 @@ export class VersionService {
     }
     return process.env.REACT_APP_CURRENT_COMMIT || "";
   }
+
+  /**
+   * from: https://stackoverflow.com/a/16427747/15155214
+   */
+  static localStorageIsAvailable() {
```
Comment (bmaden-usds) [inline frontend/src/app/VersionService.tsx:54]: Nice!

**Review by bmaden-usds — APPROVED** [review]
Thanks for investigating, fixing, and testing this!

**Pull request merged**